### PR TITLE
Update Envoy to 588d9344b31e6544869547c4bcd359b3b0f1d4cf

### PR DESCRIPTION
**Pull request by oschaaf**
- Fixes to accommodate upstream connection pool changes.
- Fixes to accommodate upstream cluster related changes.

Signed-off-by: Otto van der Schaaf <oschaaf@we-amp.com>

### CODE REVIEW

**Comment by oschaaf** [comment]
This seems to run into a challenge: Envoy deprecated the v2 api, which means we either need to 
- set flags to allow continued use of the v2 bootstrap / configuration format, which I think will be possible up to Q1 2021
- update configurations used in test and published in docs/images
- add code to automatically translate to control timing of these changes

File: source/server/README.md
Comment (mum4k) [inline source/server/README.md:37]: Was this automatically generated or does it need a merge? Not sure if we intended to have this change in this PR.

**Review by mum4k — APPROVED** [review]
Please mark as waiting for review again once we are ready to merge this in.

File: source/server/README.md
Comment (oschaaf) [reply source/server/README.md:37]: I merged master into this, it seems that `source/server/README.md` needs an update from v2 to v3. I can do that here or we can move that into a separate PR, please let me know your preference.

Comment (mum4k) [reply source/server/README.md:37]: It is fine to leave it here, I just wasn't sure it was intended since the PR description doesn't mention it. I will update it and merge. Thank you.

**Comment by mum4k** [comment]
Waiting for #584 before we merge this one in.

**Review by mum4k — APPROVED** [review]

**Pull request merged**
